.github/workflows/ruby.yml

```diff
@@ -0,0 +1,158 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+name: Ruby Agent CI
+
+on:
+  pull_request:
+    paths:
+      - 'src/ruby/**'
+      - '.github/workflows/ruby.yml'
+  push:
+    paths:
+      - 'ruby/**'
+      - '.github/workflows/ruby.yml'
+    branches:
+      - main
+    # Do not run when a tag is created.
+    tags-ignore:
+      - "**"
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  INITCONTAINER_LANGUAGE: ruby
+  K8S_OPERATOR_IMAGE_TAG: edge
+
+jobs:
+  check-modified-files:
+    name: Check whether any Ruby-related files were modified, skip the test job if not
+    uses: ./.github/workflows/check-modified-files.yml
+    secrets: inherit
+    permissions:
+      contents: read
+    with:
+      agent-language: ruby
+      
+  test:
+    name: Run Ruby init container tests
+    runs-on: ubuntu-latest
+    needs: check-modified-files
+    # run only if files were modified or the workflow was manually invoked
+    if: needs.check-modified-files.outputs.files-changed == 'true' || github.event_name == 'workflow_dispatch'
```

Comment:
You will probably want  to add `|| (github.event_name == 'release' && endsWith(github.ref_name, '_ruby')` to ensure the `tests` job runs on release.

Reply:
The `publish` job depends on the `test` job to run, but from what I understand, `publish` won't run if `test` job conditions aren't met. So this is a great call out!

Other agents appear to be missing this condition as well. I'll update Ruby's.